### PR TITLE
ACS-8121: call hxi question api

### PR DESCRIPTION
**Pull request by Pawel-608**

### CODE REVIEW

File: common-test/src/main/java/org/alfresco/hxi_connector/common/test/docker/repository/AlfrescoRepositoryContainer.java [hunk 1]
```diff
@@ -122,6 +124,7 @@ protected void configure()
                         -Dmetadata-keystore.aliases=metadata
                         -Dmetadata-keystore.metadata.password=oKIWzVdEdA
                         -Dmetadata-keystore.metadata.algorithm=DESede
-                        """.replace("\n", " "));
+                        """.replace("\n", " "))
+                .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger(AlfrescoRepositoryContainer.class.getSimpleName())));
```
Comment (krdabrowski) [inline common-test/src/main/java/org/alfresco/hxi_connector/common/test/docker/repository/AlfrescoRepositoryContainer.java:128]: This logger consumer is redundant - you can remove it.

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    private final String QUESTION_ID_ENTRY = "questionId";
+    private final int EXPECTED_STATUS_CODE = 202;
+
+    @SneakyThrows
+    public String askQuestion(Question question)
+    {
+        try (HttpEntity body = new StringEntity(objectMapper.writeValueAsString(question), APPLICATION_JSON))
+        {
+            HttpPost httpPost = new HttpPost(config.askQuestionUrl());
+            httpPost.setEntity(body);
```
Comment (krdabrowski) [inline prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java:67]: Auth header shouldn't be also passed in the request? 🤔

Comment (Pawel-608) [reply prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java:67]: yep, thats not done yet

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+
+import org.alfresco.hxi_connector.prediction_applier.service.config.QuestionServiceConfig;
+import org.alfresco.hxi_connector.prediction_applier.service.model.Question;
+
+@Slf4j
+@RequiredArgsConstructor
+public class QuestionService
```
Comment (tpage-alfresco) [inline prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java:53]: This isn't a "service" in the standard Alfresco sense - those are a layer above the DB and allow security and transaction management. Maybe we could make it clearer that this interacts with Hx Insight by calling it `HxInsightQuestionService` or `HxInsightClient` (or someone can come up with something better than these!)

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+{
+    private final QuestionServiceConfig config;
+    private final ObjectMapper objectMapper;
+    private final CloseableHttpClient client = HttpClients.createDefault();
+    private final String QUESTION_ID_ENTRY = "questionId";
+    private final int EXPECTED_STATUS_CODE = 202;
```
Comment (tpage-alfresco) [inline prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java:59]: We can use a static import to reference a named 202 constant from a dependency. ACS comes with several of these iirc.

File: prediction-applier-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-prediction-applier-extension/alfresco-global.properties [hunk 5]
```diff
@@ -3,3 +3,14 @@
...
+hxi.auth.providers.hyland-experience.grant-type=urn:hyland:params:oauth:grant-type:api-credentials
+hxi.auth.providers.hyland-experience.client-name=acs-insight-connector
+hxi.auth.providers.hyland-experience.client-id=
+hxi.auth.providers.hyland-experience.client-secret=
+hxi.auth.providers.hyland-experience.scope=hxp.integrations
+hxi.auth.providers.hyland-experience.token-uri=http://localhost:8001/token
+hxi.auth.providers.hyland-experience.environment-key=hxi-env-key
+
+hxi.client.baseUrl=http://localhost:8001
+hxi.client.askQuestionUrl=${hxi.client.baseUrl}/v1/questions
```
Comment (tpage-alfresco) [inline prediction-applier-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-prediction-applier-extension/alfresco-global.properties:16]: I wouldn't expose `/v1/questions` - it gives us something extra to document and customers something extra to misconfigure.

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java (hunk 3 above)
Comment (Pawel-608) [reply prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java:53]: `HxInsightClient` sounds good

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java (hunk 2 above)
Comment (Pawel-608) [reply prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java]: done

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java (hunk 4 above)
Comment (Pawel-608) [reply prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/QuestionService.java]: done

File: prediction-applier-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-prediction-applier-extension/alfresco-global.properties (hunk 5 above)
Comment (Pawel-608) [reply prediction-applier-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-prediction-applier-extension/alfresco-global.properties]: done

**Review by mpichura — APPROVED** [review]
LGTM

File: prediction-applier-extension/pom.xml [hunk 6]
```diff
@@ -144,6 +144,7 @@
                                     <include>org.alfresco:alfresco-hxinsight-connector-common</include>
                                     <include>org.alfresco:alfresco-hxinsight-connector-common-authentication</include>
                                     <include>org.apache.httpcomponents.core5:httpcore5</include>
+                                    <include>org.apache.httpcomponents.client5:httpclient5</include>
```
Comment (tpage-alfresco) [inline prediction-applier-extension/pom.xml:147]: We're adding more dependencies?

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/HxInsightClient.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+@RequiredArgsConstructor
+public class HxInsightClient
+{
+    private final static String QUESTION_ID_ENTRY = "questionId";
+    private final QuestionServiceConfig config;
+    private final AuthService authService;
+    private final ObjectMapper objectMapper;
+    private final CloseableHttpClient client = HttpClients.createDefault();
+
+    @SneakyThrows
```
Comment (krdabrowski) [inline prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/HxInsightClient.java:64]: This annotation is really cool in tests, but IMO we shouldn't be using `@SneakyThrows` in production code

File: prediction-applier-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-prediction-applier-extension/context/service-context.xml [hunk 8]
```diff
@@ -121,13 +123,25 @@
         </property>
     </bean>
 
-    <bean id="hxInsightAuthClient" class="org.alfresco.hxi_connector.prediction_applier.client.HxInsightAuthClient">
+    <bean id="hxInsightAuthClient" class="org.alfresco.hxi_connector.common.adapters.auth.DefaultAuthenticationClient">
```
Comment (krdabrowski) [inline prediction-applier-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-prediction-applier-extension/context/service-context.xml:126]:
```suggestion
    <bean id="hxInsightAuthClient" class="org.alfresco.hxi_connector.prediction_applier.client.HxInsightAuthClient">
```

File: prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/HxInsightClient.java (hunk 7 above)
Comment (Pawel-608) [reply prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/HxInsightClient.java:64]: yep, you're probably right, I used it and then forgot about it

Comment (Pawel-608) [reply prediction-applier-extension/src/main/java/org/alfresco/hxi_connector/prediction_applier/service/HxInsightClient.java]: done

File: prediction-applier-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-prediction-applier-extension/context/service-context.xml (hunk 8 above)
Comment (Pawel-608) [reply prediction-applier-extension/src/main/resources/alfresco/module/alfresco-hxinsight-connector-prediction-applier-extension/context/service-context.xml]: fixed

File: prediction-applier-extension/pom.xml (hunk 6 above)
Comment (Pawel-608) [reply prediction-applier-extension/pom.xml:147]: we'll remove them in next pr

**Pull request merged**